### PR TITLE
Adding edit and delete user functionality and tests

### PR DESCRIPTION
**Pull request by Salisbum**

### CODE REVIEW

File: spec/features/user_deletes_account_spec.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
+require 'rails_helper'
+
+feature "User deletes account" do
+  scenario 'user navigates to edit account page' do
+    user_login
+    click_on 'Edit registration'
```
Comment (mlg-) [inline spec/features/user_deletes_account_spec.rb]: I like a blank line between page loads in feature tests for readability.

File: spec/features/user_deletes_account_spec.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    expect(page).to have_content('Cancel my account')
+  end
+
+  scenario 'user deletes account' do
+    user_login
+    click_on 'Edit registration'
```
Comment (mlg-) [inline spec/features/user_deletes_account_spec.rb]: Blank line here.

File: spec/features/user_deletes_account_spec.rb [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    click_on 'Cancel my account'
+
+    expect(page).to have_content('Sign Up')
+  end
+
+  scenario 'user attempts to log in with deleted account' do
+    user_login
+    click_on 'Edit registration'
```
Comment (mlg-) [inline spec/features/user_deletes_account_spec.rb]: Blank line

File: spec/features/user_deletes_account_spec.rb [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    click_on 'Cancel my account'
```
Comment (mlg-) [inline spec/features/user_deletes_account_spec.rb]: Blank line

File: spec/features/user_deletes_account_spec.rb [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+    click_on 'Login'
```
Comment (mlg-) [inline spec/features/user_deletes_account_spec.rb]: Blank line

File: spec/features/user_deletes_account_spec.rb [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+    fill_in 'Email', with: 'herpderp@gmail.com'
+    fill_in 'Password', with: '12345678'
```
Comment (mlg-) [inline spec/features/user_deletes_account_spec.rb]: Why fill in the form if you don't submit it? You could assert that the fields exist instead if that's what you're going for!

File: spec/features/user_edits_account_spec.rb [hunk 7]
```diff
@@ -0,0 +1,31 @@
+require 'rails_helper'
+
+feature "User edits account" do
+  scenario 'user navigates to edit account page' do
+    user_login
+    click_on 'Edit registration'
```
Comment (mlg-) [inline spec/features/user_edits_account_spec.rb]: Blank line

File: spec/features/user_edits_account_spec.rb [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+
+  scenario 'user edits account with correct data' do
+    user_login
+    click_on 'Edit registration'
+    fill_in 'Password', with: '11111111'
+    fill_in 'Password confirmation', with: '11111111'
+    fill_in 'Current password', with: '12345678'
+    click_on 'Update'
+
+    expect(page).to have_content('Logout')
```
Comment (mlg-) [inline spec/features/user_edits_account_spec.rb]: Is there any text indicating the change? Might be nice as an additional assert.

File: spec/features/user_edits_account_spec.rb [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+feature "User edits account" do
+  scenario 'user navigates to edit account page' do
+    user_login
+    click_on 'Edit registration'
+    expect(page).to have_content('Edit User')
+  end
+
+  scenario 'user edits account with correct data' do
+    user_login
+    click_on 'Edit registration'
```
Comment (mlg-) [inline spec/features/user_edits_account_spec.rb]: Blank line

File: spec/features/user_edits_account_spec.rb [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+    fill_in 'Password confirmation', with: '11111111'
+    fill_in 'Current password', with: '12345678'
+    click_on 'Update'
+
+    expect(page).to have_content('Logout')
+  end
+
+  scenario 'user edits account with bad data' do
+    user_login
+    click_on 'Edit registration'
```
Comment (mlg-) [inline spec/features/user_edits_account_spec.rb]: Blank line

File: spec/support/user_helpers.rb [hunk 11]
```diff
@@ -0,0 +1,8 @@
+def user_login
+  user = FactoryGirl.create(:user)
+  visit '/'
```
Comment (mlg-) [inline spec/support/user_helpers.rb]: Use `visit root_path`

File: spec/support/user_helpers.rb [hunk 12]
```diff
@@ -0,0 +1,8 @@
...
+  click_on 'Login'
+  fill_in 'Email', with: user.email
+  fill_in 'Password', with: user.password
+  click_on 'Log in'
+end
```
Comment (mlg-) [inline spec/support/user_helpers.rb]: Nice helper method!

File: spec/features/user_edits_account_spec.rb (hunk 8 above)
Comment (Salisbum) [inline spec/features/user_edits_account_spec.rb]: We have this in a PR that was just merged, this master is a few commits behind

**Comment by alxjrvs** [comment]
[selfie-0](https://github.com/thieman/github-selfies/)

**Comment by alxjrvs** [comment]
[selfie-1](https://github.com/thieman/github-selfies/)

**Pull request merged**